src/anemoi/models/preprocessing/monomapper.py

```diff
@@ -0,0 +1,129 @@
+    """Remap and convert variables for single variables."""
+
+    supported_methods = {
+        method: [f, inv]
+        for method, f, inv in zip(
+            ["log1p", "sqrt", "boxcox"],
+            [log1p_converter, sqrt_converter, boxcox_converter],
+            [expm1_converter, square_converter, inverse_boxcox_converter],
+        )
+    }
+
+    def __init__(
+        self,
+        config=None,
+        data_indices: Optional[IndexCollection] = None,
+        statistics: Optional[dict] = None,
+    ) -> None:
+        super().__init__(config, data_indices, statistics)
+        self.remappers = []
+        self.backmappers = []
+        for name, method in self.methods.items():
+            method = method or self.default
+            if method == "none":
+                continue
+            elif method in self.supported_methods:
+                self.remappers.append(self.supported_methods[method][0])
+                self.backmappers.append(self.supported_methods[method][1])
+            else:
+                raise ValueError(f"Unknown remapping method for {name}: {method}")
+        self._create_remapping_indices(statistics)
+        self._validate_indices()
+
+    def _validate_indices(self):
+        assert len(self.index_training) == len(self.index_inference) <= len(self.remappers), (
+            f"Error creating conversion indices {len(self.index_training)}, "
+            f"{len(self.index_inference)}, {len(self.remappers)}"
+        )
+
+    def _create_remapping_indices(
+        self,
+        statistics=None,
+    ):
+        """Create the parameter indices for remapping."""
+        # list for training and inference mode as position of parameters can change
+        name_to_index_training = self.data_indices.data.input.name_to_index
+        name_to_index_inference = self.data_indices.model.input.name_to_index
+        self.num_training_vars = len(name_to_index_training)
+        self.num_inference_vars = len(name_to_index_inference)
+
+        (
+            self.index_training,
+            self.index_inference,
+        ) = ([], [])
+
+        # Create parameter indices for remapping variables
+        for name in name_to_index_training:
+            method = self.methods.get(name, self.default)
+            if method == "none":
+                continue
+            elif method in self.supported_methods:
+                self.index_training.append(name_to_index_training[name])
+                if name in name_to_index_inference:
+                    self.index_inference.append(name_to_index_inference[name])
+                else:
+                    # this is a forcing variable. It is not in the inference output.
+                    self.index_inference.append(None)
+            else:
+                raise ValueError[f"Unknown remapping method for {name}: {method}"]
+
+    def transform(self, x, in_place: bool = True) -> torch.Tensor:
+        for method in self.methods.values():
```

Comment:
Can we please allow to the transform and inverse_transform function not to be applied in_place?
```suggestion
        if not in_place:
            x = x.clone()   
```